### PR TITLE
feat: recreate font files from correct source (DISCO-647)

### PR DESCRIPTION
**Pull request by withoutnations**
As part of DISCO-647, this change updates the web fonts with new subsets compiled from a different set of OTF fonts. The update is to specifically fix an issue with numerals incorrectly displaying as old style figures in Milo, but also addresses missing glyphs and adds necessary font weights missing from the current assets. To ensure complete consistency, Econ Sans has also been updated with new subsets after manually checking every font file for correct glyphs.

Changes:
- Update Milo with new primary and secondary subsets compiled from the correct, current source of Milo TE
- Update Econ Sans with new primary and secondary subsets compiled from original source 
- Update Econ Sans Condensed with new primary and secondary subsets compiled from original source 
- Update MiloTE Small Caps with a new subset compiled from original source 
- Update the example file to display all above fonts
- Add support for additional weights:
   - MiloTE Medium Italic
   - MiloTE Bold Italic
   - MiloTESC (small caps) Medium Italic
   - MiloTESC (small caps) Bold Italic
   - Econ Sans OS Medium Italic
   - Econ Sans OS Bold Italic
   - Econ Sans Condensed Light Italic
   - Econ Sans Condensed Medium Italic
   - Econ Sans Condensed Bold Italic
- Refactor font-face to use font names that match the source fonts so its always clear which implementation is being used, e.g. "Milo TE" not "Milo Serif Pro"
- Update `package.json` with latest font list
- Refactor names of font-face declarations to be consistent
- Update the fallback font stacks to a modern set of system fonts
- Update the variables example in README

### CODE REVIEW

**Comment by withoutnations** [comment]
TODO: update version to 4.2.0

File: src/font-face.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+/**
+ * Econ Sans OS
+*/
 
 /* Primary subset fonts (aA-zZ 0-9 etc). */
 
 @font-face {
   font-display: swap;
-  font-family: EconSans;
+  font-family: EconSansOS;
```
Comment (fniewijk) [inline src/font-face.js]: Why not keep it the same name?

File: src/font-face.js [hunk 2]
```diff
@@ -50,25 +78,126 @@ export default `/* stylelint-disable value-keyword-case */
...
+  font-weight: 400;
   src:
     url('/assets/milo-primary-subset-rg.woff2') format('woff2'),
     url('/assets/milo-primary-subset-rg.woff') format('woff');
 }
 
 @font-face {
   font-display: swap;
-  font-family: MiloSerifPro;
+  font-family: MiloTE;
```
Comment (fniewijk) [inline src/font-face.js]: Why not keep the same name?

File: package.json [hunk 3]
```diff
@@ -17,70 +17,106 @@
...
+    "assets/econsans-condensed-secondary-subset-md-it.woff2",
+    "assets/econsans-condensed-secondary-subset-md.woff",
+    "assets/econsans-condensed-secondary-subset-md.woff2",
+    "assets/econsans-condensed-secondary-subset-rg.woff",
+    "assets/econsans-condensed-secondary-subset-rg.woff2",
+    "assets/econsans-primary-subset-bd-it.woff",
+    "assets/econsans-primary-subset-bd-it.woff2",
     "assets/econsans-primary-subset-bd.woff",
     "assets/econsans-primary-subset-bd.woff2",
-    "assets/econsans-secondary-subset-bd.woff",
```
Comment (fniewijk) [inline package.json]: Please make sure other projects don't depend on these fonts. If they do think about upping the major version of this component. eg. not from `1.0.0` to `1.0.1` but to `2.0.0`. Some project use the caret in their package.json so they might automatically update on the next build.

Example projects that might be affected: `worldif` and `worldin`.

File: src/font-face.js [hunk 4]
```diff
@@ -77,149 +206,187 @@ export default `/* stylelint-disable value-keyword-case */
...
+    url('/assets/econsans-condensed-primary-subset-bd-it.woff') format('woff');
 }
 
-/* Condensed econsans used in charts and infographics. */
+/* Secondary subset fonts (all other characters) */
 
 @font-face {
   font-display: swap;
-  font-family: EconSansCnd-secondary;
+  font-family: EconSansCndSec;
```
Comment (fniewijk) [inline src/font-face.js]: I'm all for improving the naming. It just might impact the projects that are dependent on this, they all have to be updated.

Comment (withoutnations) [reply src/font-face.js]: They've been checked and updated in FE blogs and the iOS apps; all other projects should always be using the CSS variables

File: src/font-face.js (hunk 1 above)
Comment (withoutnations) [reply src/font-face.js]: The new names are exactly the same as the source font files. For Milo, before, the CSS family name was referencing a very old source file while (mostly) using a new source file.

For Econ Sans, there are two variants of the font: Econ Sans OS and Econ Sans. By being specific, we know which variant we are using for the web

File: src/font-face.js (hunk 2 above)
Comment (withoutnations) [reply src/font-face.js]: https://github.com/economist-components/component-typography/pull/119#discussion_r211899635

File: src/font-family.js [hunk 5]
```diff
@@ -1,15 +1,38 @@
+const sans = [
```
Comment (dosyara) [inline src/font-family.js]: `sans` and `serif` names might be somewhat misleading. maybe `systemSans` or similar?

File: src/font-family.js [hunk 6]
```diff
@@ -1,15 +1,38 @@
...
     // for backwards compatibility only
     prefix: '--fontfamily',
-    'sans-default': '\'EconSansOS\', \'EconSansOSSec\', -apple-system, BlinkMacSystemFont, \'Segoe UI\', Roboto, Helvetica, Arial, sans-serif,',
-    'sans': 'var(--fontfamily-sans-custom, var(--fontfamily-sans-default))',
-    'serif-default': '\'MiloTE\', \'MiloTESec\', Constantia, \'Lucida Bright\', Lucidabright, \'Lucida Serif\', Lucida, \'DejaVu Serif\', \'Bitstream Vera Serif\', \'Liberation Serif\', Georgia, serif',
-    'serif': 'var(--fontfamily-serif-custom, var(--fontfamily-serif-default))',
-    'sans-condensed-default': '\'EconSansCnd\', \'EconSansCndSec\', -apple-system, BlinkMacSystemFont, \'Segoe UI\', Roboto, Helvetica, Arial, sans-serif,',
-    'smallcaps-serif-default': '\'MiloTESC\', Constantia, \'Lucida Bright\', Lucidabright, \'Lucida Serif\', Lucida, \'DejaVu Serif\', \'Bitstream Vera Serif\', \'Liberation Serif\', Georgia, serif',
+    'sans-default': `\'EconSansOS\', '\'EconSansOSSec\', ${ sans }`,
+    'sans': 'var(--fontfamily-sans-custom), var(--fontfamily-sans-default)',
```
Comment (dosyara) [inline src/font-family.js]: apparently I was wrong. 
`--fontfamily-sans: var(--fontfamily-sans-custom, var(--fontfamily-sans-default));` – was in original file, so we should not change this, I think

Comment (dosyara) [reply src/font-family.js]: same in serif

**Pull request merged**